python_modules/dagster/dagster_tests/definitions_tests/asset_reconciliation_tests/asset_reconciliation_scenario.py

```diff
@@ -201,6 +204,113 @@ def test_time_fn():
 
         return run_requests, cursor
 
+    def do_daemon_scenario(self, instance, scenario_name=None):
```

Comment:
It seems like there's a lot of duplicated code between this an `do_scenario` - is there a way to consolidate? Also, should we rename `do_scenario` to `do_sensor_scenario`?

Reply:
pulled the workspace generation out to a helper at least